onnxruntime/core/providers/qnn/builder/opbuilder/slice_op_builder.cc

```diff
@@ -46,19 +47,36 @@ class SliceOpBuilder : public BaseOpBuilder {
 
 Status SliceOpBuilder::ExplictOpCheck(QnnModelWrapper& qnn_model_wrapper, const NodeUnit& node_unit) const {
   size_t input_count = node_unit.Inputs().size();
-  ORT_RETURN_IF_NOT(input_count >= SLICE_MIN_INPUT && input_count <= SLICE_MAX_INPUT,
-                    "For ONNX Slice operation the expected number of inputs is between 3 and 5.");
-  // Skip the first input. All other input need to be initializer
-  for (size_t i = 1; i < node_unit.Inputs().size(); i++) {
-    const auto& next_input = node_unit.Inputs()[i].node_arg.Name();
-    if (!qnn_model_wrapper.IsInitializerInput(next_input)) {
-      return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "QNN desn't support dynamic slice.");
+  // Op set 9 only has 1 input with starts, ends, axes attribute
+  // Op set > 9, starts, ends, axes are from node input
+  if (input_count > 1) {
+    // Skip the first input. All other input need to be initializer
+    for (size_t i = 1; i < input_count; i++) {
+      const auto& next_input = node_unit.Inputs()[i].node_arg.Name();
+      if (!qnn_model_wrapper.IsInitializerInput(next_input)) {
+        return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "QNN desn't support dynamic slice.");
+      }
     }
   }
 
   return Status::OK();
 }
 
+void SliceOpBuilder::GetDataFromAttribute(const NodeUnit& node_unit,
+                                          TensorShapeVector& raw_starts,
+                                          TensorShapeVector& raw_ends,
+                                          TensorShapeVector& raw_axes) const {
+  NodeAttrHelper node_helper(node_unit);
+  auto starts = node_helper.Get("starts", std::vector<int64_t>{0});
+  raw_starts.assign(starts.begin(), starts.end());
+  auto ends = node_helper.Get("ends", std::vector<int64_t>{0});
+  raw_ends.assign(ends.begin(), ends.end());
+  if (node_helper.HasAttr("axes")) {
```

Comment:
If the `axes` attribute is not present, the ONNX spec states that the default value should be `[0, 1, …, len(starts) - 1]`. This can be added to the else path.

Reply:
It's handled by code line 183

---
In reply to: [1147319226](https://github.com/microsoft/onnxruntime/pull/15186#discussion_r1147319226) [](ancestors = 1147319226)